crates/optimism/payload/src/builder.rs

```diff
@@ -193,7 +193,7 @@ where
     // so we can safely assume that this will always be triggered upon the transition and that
     // the above check for empty blocks will never be hit on OP chains.
     reth_evm_optimism::ensure_create2_deployer(
-        chain_spec.clone(),
+        Arc::new((*chain_spec).clone().into()),
```

Comment:
this is very expensive and this perf critical, could we make this fn accept the op chainspec instead?

Reply:
I know and I'm not sure there is a real solution here because:
- `ensure_create2_deployer` is used at 2 different places now:
    - One here where we are using `OpEvmExecutor.chain_spec` which is now an Arc over an `OpChainSpec`
https://github.com/paradigmxyz/reth/blob/875cee27aca747315af1ef2a9dcdf29e058b33e7/crates/optimism/evm/src/execute.rs#L142 
  - One at this position you mention in the code where we have to do the conversion to transform the `ChainSpec` obtained via `PayloadConfig` to an `OpChainSpec`


So in my opinion, if we consider that this PR has a place to be of course, we will have to make the conversion in one of the two uses necessarily no?